src/installer.ts

```diff
@@ -338,10 +365,56 @@ async function resolveVersionFromManifest(
   }
 }
 
+function evaluateNightlyVersions(
+  versions: string[],
+  versionSpec: string
+): string {
+  let version = '';
+  let range: string | null | undefined;
+  const [raw, prerelease] = versionSpec.split('-');
+  const isValidVersion = semver.valid(raw);
+  const rawVersion = isValidVersion ? raw : semver.coerce(raw);
+  if (rawVersion) {
+    if (prerelease !== 'nightly') {
+      range = `${rawVersion}-${prerelease.replace('nightly', 'nightly.')}`;
+    } else {
+      range = `${semver.validRange(`^${rawVersion}-0`)}-0`;
+    }
+  }
+
+  if (range) {
+    versions.sort((a, b) => +semver.lt(a, b) * 1 - 0.5);
```

Comment:
Unnecessary operation "* 1" because it is already being cast to a number by the "+" operator
```suggestion
    versions.sort((a, b) => +semver.lt(a, b) - 0.5);
```